### ISSUE TITLE
schedule selector: repeat parameter is ignored, always repeats forever

### ISSUE THREAD

**Issue by LearnCocos2D** [cocos2d/cocos2d-objc#590]
Tested with develop-v3, commit: a2e01590c770fcee54ebb8c8d90ba72a8bdaf9c4

The schedule with repeat CCNode method ignores the repeat parameter, instead it always sets the timer to repeat forever:

```
-(CCTimer *) schedule:(SEL)selector interval:(CCTime)interval repeat: (uint) repeat delay:(CCTime) delay
{
    //...
    timer.repeatCount = CCTimerRepeatForever;
    //...
}
```

Moreover, the repeat type is incorrect, should be NSUInteger instead of uint to avoid compiler warnings respectively make it possible to pass in a repeat of value CCTimerRepeatForever (NSUIntegerMax).

Lastly there's one usage of kCCRepeatForever defined as: (UINT_MAX -1). Used by:
-(CCTimer *) schedule:(SEL)selector interval:(CCTime)interval

It should be replaced with CCTimerRepeatForever because they are different constants, probably causing timers not to repeat forever if the core issue (above) is being fixed.

**Comment by LearnCocos2D** [comment]
Note: this issue is also responsible for scheduleOnce: calling the selector not once but repeatedly.

**Comment by slembcke** [comment]
That was fixed a while ago in eb6b64d6e327. Are you certain you have the right branch?

**Comment by LearnCocos2D** [comment]
Sorry, apparently not. I tested the code in the project I had created before updating the cocos2d templates. So I had two almost identical projects open in Xcode and was testing with the wrong one.

Confirmed that this is working as intended in the latest version. Only the (uint) parameter needs to be fixed.
